File: compiler/bench/whileloop3.fs
\ expected: 1000000000
\ Nested WHILE 2 deep, 100K x 10K
: main
  0 0
  begin
    dup 100000 <
  while
    0
    begin
      dup 10000 <
    while
      rot 1+ -rot
      1+
    repeat
    drop
    1+
  repeat
  drop . cr
;

File: compiler/bench/whileloop3.c
// expected: 1000000000
// Nested WHILE 2 deep, 100K x 10K
#include <stdio.h>

int main(void) {
    long sum = 0;
    long i = 0;
    while (i < 100000) {
        long j = 0;
        while (j < 10000) {
            sum++;
            j++;
        }
        i++;
    }
    printf("%ld\n", sum);
    return 0;
}
